backend/dataall/modules/redshift_datasets_shares/services/redshift_table_share_processor.py

```diff
@@ -357,20 +398,32 @@ def process_revoked_shares(self) -> bool:
                         log.info(
                             f'No other tables of this dataset are shared with this namespace {self.target_connection.nameSpaceId}'
                         )
-                        redshift_client_in_target.drop_database(database=local_db)
-                        redshift_client_in_source.drop_datashare(datashare=self.datashare_name)
+                        self.redshift_data_client_in_target.drop_database(database=self.local_db)
+                        self.redshift_data_client_in_source.drop_datashare(datashare=self.datashare_name)
+
+                    # Update NON-FAILED tables with Success Action (Revoke_In_Progress ---> Revoke_Succeeded)
+                    non_failed_item_SM = ShareItemSM(started_state)
+                    final_state = non_failed_item_SM.run_transition(ShareItemActions.Success.value)
+                    non_failed_item_SM.update_state(self.session, self.share.shareUri, final_state)
+                    for table in self.tables:
+                        share_item = ShareObjectRepository.find_sharable_item(
+                            self.session, self.share.shareUri, table.rsTableUri
+                        )
+                        ShareStatusRepository.update_share_item_health_status(
+                            self.session, share_item, None, None, share_item.lastVerificationTime
+                        )
 
             except Exception as e:
                 log.error(
                     f'Failed to clean up shared resources in redshift datashares for redshift dataset {self.dataset.name} '
                     f'tables {[t.name for t in self.tables]} '
-                    f'from source {self.source_connection.name} in namespace {self.source_connection.nameSpaceId}'
-                    f'with target {self.target_connection.name} in namespace {self.target_connection.nameSpaceId}'
+                    f'from source {self.source_connection.name} in namespace {self.source_connection.nameSpaceId} '
+                    f'with target {self.target_connection.name} in namespace {self.target_connection.nameSpaceId} '
                     f'due to: {e}'
                 )
                 if not self.reapply:
-                    new_state = revoked_item_SM.run_transition(ShareItemActions.Failure.value)
-                    revoked_item_SM.update_state(self.session, self.share.shareUri, new_state)
+                    all_failed_state = revoked_item_SM.run_transition(ShareItemActions.Failure.value)
+                    revoked_item_SM.update_state(self.session, self.share.shareUri, all_failed_state)
```

Comment:
Not possible to have `self.reapply` be `True` for revoking shares items

Think we can safely always execute L425 + 426

Reply:
Modified